exec/java-exec/src/main/java/org/apache/drill/exec/work/batch/BaseRawBatchBuffer.java

```diff
@@ -172,7 +172,7 @@ public RawFragmentBatch getNext() throws IOException {
     } catch (final InterruptedException e) {
 
       // We expect that the interrupt means the fragment is canceled or failed, so we should kill this buffer
-      if (!context.shouldContinue()) {
+      if (!context.getExecutorState().shouldContinue()) {
```

Comment:
ExecutorState is an interface. I removed those methods from the FragmentContext interface because it felt redundant to replicate the ExecutorState interface methods in the FragmentContext interface.